modules/bitcoind.nix

```diff
@@ -328,21 +338,22 @@ in {
           ${extraRpcauth}
           ${/* Enable bitcoin-cli for group 'bitcoin' */ ""}
           printf "rpcuser=${cfg.rpc.users.privileged.name}\nrpcpassword="; cat "${secretsDir}/bitcoin-rpcpassword-privileged";
+          echo
+          ${optionalString (cfg.getPublicAddressCmd != "") ''
+            echo "externalip=$(${cfg.getPublicAddressCmd})"
+          ''}
         )
         confFile='${cfg.dataDir}/bitcoin.conf'
         if [[ ! -e $confFile || $cfg != $(cat $confFile) ]]; then
           install -o '${cfg.user}' -g '${cfg.group}' -m 640 <(echo "$cfg") $confFile
         fi
       '';
-      postStart = ''
-        # Poll until bitcoind accepts commands. This can take a long time.
-        while ! ${cfg.cli}/bin/bitcoin-cli getnetworkinfo &> /dev/null; do
-          sleep 1
-        done
-      '';
       serviceConfig = nix-bitcoin-services.defaultHardening // {
+        Type = "notify";
+        NotifyAccess = "all";
```

Comment:
Just for my understanding, why isn't the default "main" enough?

Reply:
I've also wondered why that's the case, because bitcoind's call to `notify` should happen from the main cgroup.
But experimentally only value `all` works, and I didn't feel like digging into this.

Reply:
Okay

Reply:
FTR I had troubles with it as well and I had to move `systemd-notify` from a separate script to the config directly.

If you are willing to give it one more attempt maybe put `exec` before `systemd-notify` to replace the `sh` process created by `bitcoind` instead of creating yet another child. I didn't try yet, just suspicion.

Reply:
Hah, it seems we've all misread the docs.

Here's the full story:
```
Value | Process(es) allowed to notify
------+--------------------------------------------------------------
main  | The process launched by ExecStart
exec  | The process launched by ExecStart{,Pre,Post,...}
all   | Processes in the cgroups launched by ExecStart{,Pre,Post,...}
```
So when a service uses systemd's `notify` helper binary instead of directly writing to the notify socket, it will always need to use value `all`.